Handle filtering by name and minor exceptions

kubealex opened the pull request:
hi @sabre1041, first of all thank you for your effort, with this PR I would like to address some issues I faced:

- There's a failure when using "name" as a parameter due to python kubernetes-client returning a named object [1]. Unfortunately the PR with the fix is not yet merged, so I introduced the following changes:

[v] Add the name as a field_selector, appending it if there are other selectors.
[v] Since the "name" parameter is handled differently, I removed it from the list when enriching the client options

- Wrapped the event loop in a try/catch to handle generic connectivity exceptions (API server unavailability, etc) to avoid the rulebook to exit

- Added the field_selectors documentation in the README

Now the flow is smooth,  I tested it the whole morning with different combinations and kept it running, and it's still working like a charm.

Hope this is useful,
Ale

[1] https://github.com/kubernetes-client/python/pull/2076

kubealex committed:
Added field-selector to doc

kubealex committed:
fix using "name" as parameter

sabre1041 reviewed (approved):
LGTM. Thanks @kubealex !

sabre1041 reviewed (changes requested):
welp... jumped the gun (gotta love CI). Can you take a look at the errors?

kubealex committed:
CI fixes and version bump

kubealex commented:
done @sabre1041, thank you! They were not strictly linked to the code, to summarize, they were related to ansible-lint complaining about:

- Missing CHANGELOG.md, I added an empty one.
- Ansible-core version in runtime.yml was 2.11, minimum supported version according to ansible-lint is 2.13.0 [1] so I edited that accordingly.

I also bumped the version to 1.0.1 in galaxy.yml to avoid the release workflow failing due to duplicated versions (feel free to edit that based on how you want to version this) and squashed my commits to clearly identify the different fixes.

[1] https://github.com/ansible/ansible-lint/blob/main/src/ansiblelint/rules/meta_runtime.md

sabre1041 reviewed (approved):
LGTM
